File: battery_life.py
import argparse
from datetime import date, timedelta
from sys import argv, exit, stdout

start = date(2017, 1, 10)
today = date.today()
max_cycles = 1000


def battery_life(cycles):
    """Calculates expected day to reach 1000 cycles assuming `start` is the
    first day of use.
    """
    final = start + timedelta(days=max_cycles / (cycles / (today - start).days))
    return final.strftime("%d %h %Y")


if __name__ == "__main__":
    parser = argparse.ArgumentParser(description="Battery life utility.")
    parser.add_argument(
        "--output",
        "-o",
        nargs="?",
        type=argparse.FileType("a+"),
        default=stdout,
        help="Specifies output file.",
    )
    parser.add_argument("--cycles", type=str, help="Battery cycle count.")
    parser.add_argument("--capacity", type=str, help="Full battery capacity.")
    parser.add_argument("--design-capacity", type=str, help="Design battery capacity.")

    if len(argv) < 2:
        parser.print_help()
        exit(1)

    args = parser.parse_args()

    data = {
        "date": today.strftime("%d %h %Y"),
        "cycles": args.cycles if args.cycles else "",
        "final": battery_life(int(args.cycles)) if args.cycles is not None else "",
        "capacity": args.capacity if args.capacity else "",
        "design_capacity": args.design_capacity if args.design_capacity else "",
    }

    if args.output is not stdout:
        with open(args.output.name, "a+") as f:
            f.write(",".join(data.values()) + "\n")
    else:
        print(f"".join(f"{k}" for k in data))
        print(f"".join(f"{v}" for v in data.values()))

File: batteryLife.js
// JavaScript version

const msPerDay = 1000 * 60 * 60 * 24;
const maxCycles = 1000; 
const start = new Date(2017, 0, 11);
const today = new Date();

Date.prototype.addDays = function(days) {
    var date = new Date(this.valueOf());
    date.setDate(date.getDate() + days);
    return date;
}

function batteryLife(cycles) {
    const daysFromStart = Math.round((today - start) / msPerDay, 0);
   
    const maxDays = Math.round(maxCycles/(cycles/daysFromStart), 0);

    var futureDate = start.addDays(maxDays);

    return futureDate
}

futureDate = batteryLife(parseInt(process.argv[process.argv.length-1])).toDateString();
console.log('Expected day for ' + maxCycles + ' cycles: ' + futureDate + '.')


